src/app.jsx

```diff
@@ -196,59 +253,78 @@ class Application extends React.Component {
+            this.handleImageEvent(event, root);
             break;
         default:
             console.warn('Unhandled event type ', event.type);
         }
     }
 
-    init() {
-        utils.podmanCall("GetVersion")
+    init(root) {
+        this.checkUserService();
+        utils.podmanCall("GetVersion", {}, root)
                 .then(reply => {
-                    this.setState({ serviceAvailable: true });
-                    this.updateImagesAfterEvent();
-                    this.updateContainersAfterEvent();
+                    this.setState({ [root ? "rootServiceAvailable" : "userServiceAvailable"]: true });
+                    this.updateImagesAfterEvent(root);
+                    this.updateContainersAfterEvent(root);
                     utils.monitor("GetEvents", {},
                                   message => {
-                                      message.parameters && message.parameters.events && this.handleEvent(message.parameters.events);
-                                  },
-                                  () => {
-                                      this.setState({ serviceAvailable: false });
+                                      message.parameters && message.parameters.events && this.handleEvent(message.parameters.events, root);
+                                  }, isRoot => {
+                                      this.setState({ [isRoot ? "rootServiceAvailable" : "userServiceAvailable"]: false
+                                      });
                                   },
+                                  root
                     );
                 })
                 .catch(error => {
-                    if (error.name === "ConnectionClosed")
-                        this.setState({ serviceAvailable: false });
-                    else
+                    if (error.name === "ConnectionClosed") {
+                        this.setState({ [root ? "rootServiceAvailable" : "userServiceAvailable"]: false,
+                                        [root ? "rootContainersLoaded" : "userContainersLoaded"]: true,
+                                        [root ? "rootImagesLoaded" : "userImagesLoaded"]: true
+                        });
+                    } else
                         console.error("Failed to call GetVersion():", error);
                 });
     }
 
     componentDidMount() {
-        this.init();
+        this.init(true);
+        this.init(false);
+    }
+
+    checkUserService() {
+        let argv = ["systemctl", "--user", "status", "io.podman.socket"];
+
+        cockpit.spawn(argv)
+                .then(() => this.setState({ userServiceExists: true }))
+                .catch(err => {
+                    if (err.exit_status === 4) // systemctl status returns 3 when service is not running
+                        this.setState({ userServiceExists: false });
+                    else
+                        this.setState({ userServiceExists: true });
```

Comment:
It seems a little bold to assume that any code except 4 means success. E. g. in the "not running" case, setting it to true still means that you will try and talk to it. It seems a bit more robust to check if the command succeeds, and assume "not available" otherwise.